compiler/rustc_middle/src/ty/context.rs

```diff
@@ -3140,6 +3141,35 @@ impl<'tcx> TyCtxt<'tcx> {
             && self.constness(def_id) == hir::Constness::Const
     }
 
+    pub fn enforce_trait_const_stability(self, trait_def_id: DefId, span: Span) {
+        match self.lookup_const_stability(trait_def_id) {
+            Some(ConstStability {
+                level: StabilityLevel::Unstable { implied_by: implied_feature, .. },
+                feature,
+                ..
+            }) => {
+                if span.allows_unstable(feature)
+                    || implied_feature.is_some_and(|f| span.allows_unstable(f))
+                {
+                    return;
+                }
+                let feature_enabled = trait_def_id.is_local()
+                    || self.features().enabled(feature)
+                    || implied_feature.is_some_and(|f| self.features().enabled(f));
+
+                if !feature_enabled {
+                    let mut diag = self.dcx().create_err(crate::error::UnstableConstTrait {
+                        span,
+                        def_path: self.def_path_str(trait_def_id),
+                    });
+                    self.disabled_nightly_features(&mut diag, None, [(String::new(), feature)]);
+                    diag.emit();
+                }
+            }
+            _ => {}
```

Comment:
Is it guaranteed that there's a const stability attribute if it's in a staged_api crate?

Reply:
Ah lmao, one scroll further down is the answer